src/components/Button/Button.tsx

```diff
@@ -0,0 +1,164 @@
+   */
+  children?: ReactNode,
+
+  /**
+   * Defines the button hierarchy. Either:
+   *
+   * - primary: button associated with the most significant (and therefore primary) action on the page;
+   * - neutral: button associated with a general purpose action;
+   * - danger: button associated with a potentially dangerous action.
+   */
+  hierarchy?: ButtonHierarchies,
+
+  /**
+   * Redirect url of a link button.
+   */
+  href?: string,
+
+  /**
+   * Sets an icon component for the button.
+   */
+  icon?: ReactNode,
+
+  /**
+   * Defines a position for the button icon (if any). Either:
+   *
+   * - left: the icon is placed to the left of the button text;
+   * - right: the icon is placed to the right of the button text.
+   */
+  iconPosition?: ButtonIconPositions,
+
+  /**
+   * Identifies whether the button is disabled or not.
+   */
+  isDisabled?: boolean,
+
+  /**
+   * Sets the loading status of the button.
+   */
+  isLoading?: boolean,
+
+  /**
+   * Sets the handler to handle a click event.
+   */
+  onClick?: () => void,
+
+  /**
+   * Defines the button shape. Either:
+   *
+   * - square: square-shaped button with rounded corners;
+   * - circle: fully rounded button, typically used for buttons containing only icon.
+   */
+  shape?: ButtonShapes,
+
+  /**
+   * Defines the button size. Either:
+   *
+   * - small: small button - 24px in height;
+   * - middle: middle button - 32px in height;
+   * - large: large button - 48px in height.
+   */
+  size?: ButtonSizes,
+
+  /**
+   * Specifies where the linked document will open when the link is clicked.
+   * Only usable for link buttons together with the href property.
+   *
+   * @link https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a
+   */
+  target?: string,
+
+  /**
+   * Defines the button type. Either:
+   *
+   * - filled: button with a solid fill, typically used for primary actions;
+   * - outlined: medium-emphasis button with a stroke around the button container and no fill,
+   *   typically used for secondary actions;
+   * - ghost: transparent button with no stroke, typically used for links.
+   */
+  type?: ButtonTypes,
+}
+
+/**
+ * UI component for presenting buttons
+ *
+ * @link https://ant.design/components/button
+ * @returns {Button} Button component
+ */
+export const Button = ({
+  children,
+  hierarchy,
+  href,
+  icon,
+  iconPosition,
+  isDisabled,
+  isLoading,
+  onClick,
+  shape,
+  size,
+  target,
+  type,
+}: ButtonType): ReactElement => {
+  const buttonClassNames = useMemo(() => (
+    [
+      button,
+      children && size === Small && buttonSm,
+      children && size === Middle && buttonMd,
+      children && size === Large && buttonLg,
+      type === Ghost && buttonGhost,
+    ].filter(Boolean)
+  ), [children, size, type])
+
+  /**
+   * rel: 'noopener noreferrer'
+   *
+   * @link https://github.com/ant-design/ant-design/blob/master/components/button/button.tsx#L259
+   */
```

Comment:
What is the purpose of this comment? I don't think it is accessible for developers on hover if we put it inside the component. I would rather add details to the href property.

Moreover, I visited the link but I found it quite difficult to grasp the meaning of its content.

Reply:
We chose to put that link because it represents the only reference "documentation" for the fact that we put `rel` between the props of the Ant component. The Ant button does not document `rel` among the possible props, but in fact, accepts it and makes the spread within an HTML component of type `<a>`, as seen in the link.
I can move the link to the `href` section and try to provide this explanation there as well, although it seems slightly out of scope

Reply:
We add that ref works because in the case of prop href and uses an html <a> tag

Reply:
I think a person without context can have some difficulties understanding that. I would try to add some explanation details to make it more clear. 

However, I don't think this comment is necessary at all. As long as the Ant Button component type does not raise errors on such a property, I think we can conclude it is actually supported. 
In addition, the rel attribute is also very harmless. In general, applying a rel attribute on HTML tags that do not accept it represents more of a bad practice than a blocking error that can break the dom.

Reply:
I removed the comment and added in the documentation on `href` the following sentence: `For security reasons, the attribute "rel: 'noopener noreferrer'" is always specified.`
I think it's the best compromise 🤝🏼🤝🏼